### PR TITLE
feat(api): add POST method that will add a uuid

### PR DESCRIPTION
**Pull request by jasongoodwin**
(Intended to be published at 0.2.0).

This adds a `POST` method that will append `/$UUID` to the path.
It vendorizes the google uuid plugin as required by traefik.
Not sure if it works (how do I test it?), looks okay tho! We added it to Yoel's repo.

### CODE REVIEW

**Comment by jasongoodwin** [comment]
Francis made a good observation that the generated uuid isn't returned to the client. Looking.

**Comment by jasongoodwin** [comment]
Okay added a header that will return the ObjectLocation. Need to figure out how to test it - we can deploy/test by hand but if any guidance lmk.

File: README.md [hunk 1]
```diff
@@ -37,7 +36,9 @@ To store objects in a local directory, use the following labels (example):
 "traefik.http.middlewares.my-aws.plugin.aws.directory" : "aws-local-directory"
 ```
 
-`GET` and `PUT` are supported.
+`GET`, `PUT` and `POST` are supported.
+`POST` will append a UUID to the path. There is an `ObjectLocation` header in the response.
```
Comment (francisf2) [inline README.md:40]: Unless there's a precedent for using `ObjectLocation` as the header name, I would just use `Location`, which is more standard.

Comment (jasongoodwin) [reply README.md:40]: Sure! Object is the s3 terminology. I'll change it tho.

File: README.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ To store objects in a local directory, use the following labels (example):
 ```
 
 `GET`, `PUT` and `POST` are supported.
-`POST` will append a UUID to the path. There is an `ObjectLocation` header in the response.
+`POST` will append a UUID to the path. There is an `Location` header in the response.
```
Comment (francisf2) [inline README.md:40]: an `Location` -> a `Location` (same below)

File: README.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ To store objects in [Amazon Simple Storage Service (S3)](https://docs.aws.amazon
 Note that `prefix` must include the leading slash.
 
 [PUT](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html) and [GET](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html) are supported. 
-If you make a POST request, a UUID will be generated for the object key and will put that object. There is an `ObjectLocation` header in the response.
+If you make a POST request, a UUID will be generated for the object key and will put that object. There is an `Location` header in the response.
```
Comment (francisf2) [inline README.md:58]: I would rephrase 'and will put that object' -> 'and the object will be created in the same manner as a PUT request'

**Review by francisf2 — APPROVED** [review]

**Comment by jasongoodwin** [comment]
Linting fails on this, can't reproduce locally. Looking... 

Here is from my local run.

``` 
% golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/jgoodwin/Development/src/new-traefik-plugin/mine/traefik-aws-plugin /Users/jgoodwin/Development/src/new-traefik-plugin/mine /Users/jgoodwin/Development/src/new-traefik-plugin /Users/jgoodwin/Development/src /Users/jgoodwin/Development /Users/jgoodwin /Users /]
INFO [lintersdb] Active 6 linters: [errcheck gosimple govet ineffassign staticcheck unused]
INFO [loader] Go packages loading at mode 575 (types_sizes|compiled_files|deps|exports_file|imports|name|files) took 357.170458ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 1.086291ms
INFO [linters_context/goanalysis] analyzers took 0s with no stages
INFO [runner] processing took 1.208µs with stages: max_same_issues: 333ns, skip_dirs: 167ns, nolint: 166ns, cgo: 84ns, identifier_marker: 42ns, path_prettifier: 42ns, autogenerated_exclude: 42ns, max_from_linter: 42ns, exclude: 42ns, max_per_file_from_linter: 42ns, skip_files: 42ns, severity-rules: 41ns, fixer: 41ns, diff: 41ns, uniq_by_line: 41ns, path_shortener: 0s, exclude-rules: 0s, source_code: 0s, path_prefixer: 0s, sort_results: 0s, filename_unadjuster: 0s
INFO [runner] linters took 199.164583ms with stages: goanalysis_metalinter: 199.14375ms
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 7 samples, avg is 28.2MB, max is 35.9MB
INFO Execution took 566.039667ms
jgoodwin@MP16JGOODWIN1 traefik-aws-plugin % echo $?
0
```

**Comment by jasongoodwin** [comment]
Looks like the vendored packages have to be installed in the GOPATH for yaegi to pick them up for interpreting go.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -35,6 +35,14 @@ jobs:
     - name: Set up GOPATH
       run: go env -w GOPATH=${{ github.workspace }}/go
 
+
+      - name: Check and get dependencies
```
Comment (francisf2) [inline .github/workflows/build.yaml:39]: Offset seems wrong

Comment (jasongoodwin) [reply .github/workflows/build.yaml]: Thanks - Fixed. was hoping to iterate on it on my fork first. Don't have admin access but I think there is a toggle to allow forks to run the workflow.

**Comment by yoeluk** [comment]
LGTM

**Comment by yoeluk** [comment]
@francisf2 @ilaidlaw  can I have write access to this project?

**Review by francisf2 — APPROVED** [review]
